lib/booking/ui/calendar.dart

```diff
@@ -208,41 +166,271 @@ class Calendar extends HookConsumerWidget {
   }
 }
 
-_AppointmentDataSource _getCalendarDataSource(List<Booking> res) {
-  List<Appointment> appointments = <Appointment>[];
-  res.map((e) {
-    if (e.recurrenceRule != "") {
-      final dates = getDateInRecurrence(e.recurrenceRule, e.start);
-      dates.map((data) {
-        appointments.add(Appointment(
-          startTime: combineDate(data, e.start),
-          endTime: combineDate(data, e.end),
-          subject: '${e.room.name} - ${e.reason}',
-          isAllDay: false,
-          startTimeZone: "Europe/Paris",
-          endTimeZone: "Europe/Paris",
-          notes: e.note,
-          color: generateColor(e.room.name),
-        ));
-      }).toList();
-    } else {
-      appointments.add(Appointment(
-        startTime: e.start,
-        endTime: e.end,
-        subject: '${e.room.name} - ${e.reason}',
-        isAllDay: false,
-        startTimeZone: "Europe/Paris",
-        endTimeZone: "Europe/Paris",
-        notes: e.note,
-        color: generateColor(e.room.name),
-      ));
-    }
-  }).toList();
-  return _AppointmentDataSource(appointments);
+class _AppointmentDataSource extends CalendarDataSource<Booking> {
```

Comment:
The standard is to make a file for each class.
You can remove the underscore when moving it